os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -201,6 +203,16 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 		return -ENOMEM;
 	}
 #endif
+#endif
+
+#if defined(CONFIG_DEBUG_MM_HEAPINFO) && defined(CONFIG_APP_BINARY_SEPARATION)
+	/* Save the text and data region size of new binary into its heap
+	 * for excluding those size when calculating the heap usage.
+	 */
+	loadinfo->binp->uheap->elf_sections_size = loadinfo->textsize + loadinfo->datasize;
```

Comment:
I think elf_sections_size is no longer required because the elf sections are not allocated from the uheap. 
The current logic is as follows:
- allocate partition for app using kumm_memalign API
- place the elf sections directly in this partition
- initialize the uheap (which is the application's heap) at the end of the data section

So, I think the elf_sections_size is no longer relevant and can be removed.